sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -665,23 +725,31 @@ def _set_dtype(self, X, centers_init):
             # X. In this case we should only warn when:
             #     (dtype == np.float64) and not self.has_aspect_fp64
             # The other cases would not trigger any additional memory copies.
-            X = X.astype(dtype)
+            X = X.astype(compute_dtype)
 
         centers_init_dtype = centers_init.dtype
-        if centers_init.dtype != dtype:
+        if centers_init.dtype != compute_dtype:
             warnings.warn(
-                f"The centers have been initialized with type {centers_init_dtype} but "
-                f"type {dtype} is expected. A copy will be created with the correct "
-                f"type {dtype}. Ensure that the centers are initialized with the "
-                f"correct dtype to save memory and disable this warning.",
+                f"The centers have been passed with type {centers_init_dtype} but "
+                f"type {compute_dtype} is expected. A copy will be created with the "
+                f"correct type {compute_dtype}. Ensure that the centers are passed "
+                f"with the correct dtype to save memory and disable this warning.",
                 DataConversionWarning,
             )
-            centers_init = centers_init.astype(dtype)
+            centers_init = centers_init.astype(compute_dtype)
 
-        return X, centers_init, dtype
+        return X, centers_init, compute_dtype, output_dtype
 
-    def _check_inputs(self, X, cluster_centers):
-        X, cluster_centers, dtype = self._set_dtype(X, cluster_centers)
+    def _check_inputs(self, X, cluster_centers, sample_weight):
+        if (sample_weight is not None) and ((sample_weight != sample_weight[0]).any()):
+            raise FeatureNotAvailableError(
+                "The sklearn_numba_dpex engine for KMeans does not support non-uniform"
+                " sample weights."
```

Comment:
Maybe we could be explicit that this is a planned feature and include the URL of an issue tracking the implementation of this feature.